server/src/controllers/maproom/v2/getArea.ts

```diff
@@ -11,43 +15,85 @@ interface Cell {
 }
 
 export const getArea: KoaController = async (ctx) => {
+  const user: User = ctx.authUser;
   const requestBody: Cell = ctx.request.body;
+  await ORMContext.em.populate(user, ["save"])
+  const save = user.save
 
   for (const key in requestBody) {
     requestBody[key] = parseInt(requestBody[key], 10) || 0;
   }
 
-  const x = requestBody.x;
-  const y = requestBody.y;
+  const currentX = requestBody.x
+  const currentY = requestBody.y
+  const width = requestBody.width || 10;
+  const height = requestBody.height || 10;
 
   // Creates {maxX} x {maxY} grid from point 0 x 0
-  const maxX = 100;
-  const maxY = 100;
+  const maxX = currentX + width;
+  const maxY = currentY + height;
+
+  const wCells = await ORMContext.em.find(WorldMapCell, {
+    x: {
+      $gte: currentX,
+      $lte: maxX,
+    },
+    y: {
+      $gte: currentY,
+      $lte: maxY,
+    },
+    world_id: "1",
+  })
+
+  const worldMap = wCells.reduce<{ [x: number]: { [y: number]: WorldMapCell } }>((acc, obj) => {
+    const { x, y } = obj;
+    if (!acc[x]) {
+      acc[x] = {};
+    }
+    acc[x][y] = obj;
+    return acc;
+  }, {});
 
   let cells = {};
 
-  for (let x = 0; x < maxX; x++) {
+  for (let x = currentX; x < maxX; x++) {
     cells[x] = {};
 
-    for (let y = 0; y < maxY; y++) {
-      cells[x][y] = await wildMonsterCell();
+    for (let y = currentY; y < maxY; y++) {
 
-      // Testing - Hardcoded co-ordinates to load base types
+      if (worldMap.hasOwnProperty(x)) {
+        if (worldMap[x].hasOwnProperty(y)) {
+          const cell = worldMap[x][y];
+          if (cell.base_type != 1) {
+            cells[x][y] = cell.uid === user.userid ? await homeCell(ctx, cell) : await enemyCell(ctx, cell)
+          } else {
+            cells[x][y] = await  wildMonsterCell(cell)
+          }
+          continue
+        }
+      }
+      // cells[x][y] = await wildMonsterCell();
+      //
+      // // Testing - Hardcoded co-ordinates to load base types
       if (x === 0 && y === 0) {
-        cells[x][y] = await homeCell(ctx);
+        // cells[x][y] = await homeCell(ctx);
+        // continue
       }
 
-      if (x === 2 && y === 1) {
-        cells[x][y] = await outpostCell(ctx);
-      }
+      cells[x][y] = await wildMonsterCell();
+      //
+      // if (x === 2 && y === 1) {
+      //   cells[x][y] = await outpostCell(ctx);
+      // }
+      // cells[x][y] = null;
```

Comment:
I will need a general overview here of what you are trying to achieve with this new code added, and the benefits of this approach.